samza-kv/src/test/java/org/apache/samza/storage/kv/TestLargeMessageSafeStore.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.storage.kv;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.samza.SamzaException;
+import org.apache.samza.serializers.JsonSerdeV2;
+import org.apache.samza.serializers.LongSerde;
+import org.apache.samza.serializers.Serde;
+import org.apache.samza.serializers.StringSerde;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class TestLargeMessageSafeStore {
+
+  @Mock
+  KeyValueStore<byte[], byte[]> store;
+  int maxMessageSize = 1024;
+  String storeName = "testStore";
+
+  @Before
+  public void setup() {
+    Mockito.doNothing().when(store).put(Matchers.any(), Matchers.any());
+    Mockito.doNothing().when(store).putAll(Matchers.any());
+  }
+
+  @Test
+  public void testLargeMessagePutWithDropLargeMessageDisabled() {
+    LargeMessageSafeStore largeMessageSafeKeyValueStore = new LargeMessageSafeStore(store, storeName, false, maxMessageSize);
+
+    byte[] key1 = new byte[16];
+    byte[] smallMessage = new byte[32];
+    largeMessageSafeKeyValueStore.put(key1, smallMessage);
+    Mockito.verify(store).put(Matchers.eq(key1), Matchers.eq(smallMessage));
```

Comment:
Can you please pull this out to a separate test case? It's easier to tell that you have a test for a small message that way.

Reply:
Done.